File: C++/Step13-Greedy Algorithm/11399.cpp
//
//  11399.cpp
//  Step13-Greedy Algorithm
//
//  Created by 김예빈 on 2019. 9. 10..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, temp, sum, i;
    
    cin >> n;
    
    vector<int> people;
    
    for(i = 0; i < n; i++) {
        cin >> temp;
        people.push_back(temp);
    }
    
    sort(people.begin(), people.end());
    sum = people.at(0);
    
    for(i = 1; i < n; i++) {
        people[i] += people[i - 1];
        sum += people[i];
    }
    
    cout << sum;
    
    return 0;
}

File: BaekJoon Online Judge/Swift/Step24-Greedy Algorithm/11399.swift
//
//  11399.swift
//  Step15-Greedy Algorithm
//
//  Created by Yebin Kim on 2019/12/19.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

if let _ = Int(readLine()!) {
    var waiting = readLine()!.split(separator: " ").map{ Int($0)! }
    waiting.sort(by: <)
    
    var sum = 0
    
    print(waiting.map({ cur -> Int in
        sum += cur
        return sum
    }).reduce(0){ $0 + $1 })
}
